### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by MariBerdnyk**
Try again

### CODE REVIEW

File: GameFighter/Battle.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿namespace GameFighter
+{
+    public static class Battle
+    {
+        public static bool Fight(Warrior warrior_1, Warrior warrior_2)
```
Comment (elkavitaly) [inline GameFighter/Battle.cs]: Don't use snake case notation

File: GameFighter/Battle.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        {
+            while(warrior_1.Alive)
+            {
+                warrior_2.Hit(warrior_1.Attack);
+                if (!warrior_2.Alive)
+                    return true;
+                warrior_1.Hit(warrior_2.Attack);
+            }
+            return false;
```
Comment (elkavitaly) [inline GameFighter/Battle.cs:29]: I would add empty line before return statement

File: GameFighter/Models/Knight.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿namespace GameFighter
+{
+    public class Knight : Warrior
+    {
+        public Knight() : base()
```
Comment (elkavitaly) [inline GameFighter/Models/Knight.cs]: You don't need to specify `base()` constructor explicitly

File: GameFighter/Models/Warrior.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+﻿namespace GameFighter
+{
+    public class Warrior
+    {
+        public int Health { get; private set; }
+        public int Attack { get; protected set; }
+        public bool Alive { get => Health > 0 ? true : false; }
```
Comment (elkavitaly) [inline GameFighter/Models/Warrior.cs]: `Health > 0` returns boolean value, so you don't need to use ternary operator here.
We discussed that it's better to use 'Is' or 'Has' prefixes for identifiers (fields, properties) which return boolean value

File: GameFighter/Models/Warrior.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
+﻿namespace GameFighter
+{
+    public class Warrior
+    {
+        public int Health { get; private set; }
```
Comment (elkavitaly) [inline GameFighter/Models/Warrior.cs]: Why the setter is private? How will you set Health property in derived classes?

File: GameFighter/Models/Warrior.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+        public int Attack { get; protected set; }
+        public bool Alive { get => Health > 0 ? true : false; }
+        public Warrior()
+        {
+            Health = 50;
+            Attack = 5;
+        }
+        public void Hit(int attack)
```
Comment (elkavitaly) [inline GameFighter/Models/Warrior.cs]: For me 'hit' means that the action is directed at someone else. In your case this means that this object is hit (passive scenario). I would rename this method or change the logic to reflect what the method actually does.

File: GameFighterXUnitTests/GameFighterTests.cs [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+        [Fact]
+        public void Attack_Warrior()
+        {
+            var warrior = new Warrior();
+            Assert.Equal(5, warrior.Attack);
+        }
+
+        [Fact]
+        public void Attack_Knight()
+        {
```
Comment (elkavitaly) [inline GameFighterXUnitTests/GameFighterTests.cs]: This test and the tests above don't test anything. You don't need to check how compiler works

File: GameFighterXUnitTests/GameFighterTests.cs [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+            var knight = new Knight();
+            Assert.Equal(7, knight.Attack);
+        }
+
+        [Fact]
+        public void Fight_Two_Warriors()
```
Comment (elkavitaly) [inline GameFighterXUnitTests/GameFighterTests.cs]: Don't use snake case notation

File: GameFighter/Models/Warrior.cs [hunk 9]
```diff
@@ -0,0 +1,18 @@
+﻿namespace GameFighter
+{
+    public class Warrior
+    {
+        public int Health { get; private set; }
+        public int Attack { get; protected set; }
+        public bool Alive { get => Health > 0 ? true : false; }
+        public Warrior()
```
Comment (elkavitaly) [inline GameFighter/Models/Warrior.cs:9]: Add extra lines between separate methods

**Review by elkavitaly — CHANGES_REQUESTED** [review]

File: GameFighter/Models/Warrior.cs (hunk 5 above)
Comment (MariBerdnyk) [reply GameFighter/Models/Warrior.cs]: I was sure, that health is a constant variable :))

File: GameFighter/Battle.cs [hunk 10]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+
+namespace GameFighter
+{
+    public static class Battle
+    {
+        public static bool Fight(Warrior warrior1, Warrior warrior2)
+        {
+            if (!warrior1.IsAlive || !warrior2.IsAlive)
```
Comment (elkavitaly) [inline GameFighter/Battle.cs:14]: What if any parameter is null?
Add curly braces for _if_ statements (here and below)

File: GameFighter/Models/Person.cs [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GameFighter.Models
+{
+    public abstract class Person
+    {
+        public int Health { get; set; }
+
+        public bool IsAlive { get => Health > 0; }
```
Comment (elkavitaly) [inline GameFighter/Models/Person.cs]: You can use more simplified syntax (lambda expression)

**Review by elkavitaly — CHANGES_REQUESTED** [review]

File: GameFighter/Models/Person.cs (hunk 11 above)
Comment (MariBerdnyk) [reply GameFighter/Models/Person.cs]: But it looks beautiful :(

File: GameFighter/Battle.cs [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+namespace GameFighter
+{
+    public static class Battle
+    {
+        public static bool Fight(Warrior warrior1, Warrior warrior2)
+        {
+            if (warrior1 == null || warrior2 == null)
+            {
+                throw new NotImplementedException("warrior1 and warrior2 must be initialised!");
```
Comment (elkavitaly) [inline GameFighter/Battle.cs]: Why do you use NotImplementedException? It's more obvious to use ArgumentNullException. Or you can combine both if statements and use InvalidArgumentException

File: GameFighterXUnitTests/GameFighterTests.cs [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+        public void Fight_KnightVsWarrior_ReturnTrue()
+        {
+            var warrior = new Warrior();
+            var knight = new Knight();
+
+            Assert.True(Battle.Fight(knight, warrior));
+        }
+
+        [Fact]
+        public void Fight_KnightVsKnight_ReternTrue()
```
Comment (elkavitaly) [inline GameFighterXUnitTests/GameFighterTests.cs]: Typo in _return_

File: GameFighterXUnitTests/GameFighterTests.cs [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+        {
+            var knight1 = new Knight();
+            var knight2 = new Knight();
+
+            Assert.True(Battle.Fight(knight1, knight2));
+        }
+
+        [Fact]
+        public void Fight_IsWarrior1AfterBattleVsWarriorAlive_ReternTrue()
```
Comment (elkavitaly) [inline GameFighterXUnitTests/GameFighterTests.cs]: I would rename this test to Fight_WarriorVsWarrior_SecondWarriorIsAlive. You don't need to ask in the test name, you should use statements

File: GameFighter/Models/Person.cs [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GameFighter.Models
+{
+    public abstract class Person
+    {
+        public int Health { get; protected set; }
+
+        public bool IsAlive
```
Comment (elkavitaly) [inline GameFighter/Models/Person.cs]: `public bool IsAlive => Health > 0` looks simpler

**Review by elkavitaly — CHANGES_REQUESTED** [review]

**Review by elkavitaly — APPROVED** [review]

**Pull request merged**